domain/machine/state/state.go

```diff
@@ -194,6 +194,31 @@ func (st *State) GetMachineLife(ctx context.Context, mName machine.Name) (*life.
 	return &lifeResult, errors.Annotatef(err, "getting life status for machines %q", mName)
 }
 
+// SetMachineLife sets the life status of the specified machine.
+// No error is returned if the provided machine doesn't exist, just nothing gets
```

Comment:
It's better that this _does_ return a not-found error.
- Make `machineLife` use a UUID instead of a name string.
- Select the UUID for the name up front and return an error if not found.
- Use that UUID in the argument if we located it.